translator/source_code/combine_wzdx.py

```diff
@@ -0,0 +1,177 @@
+import json
+import copy
+from shapely.geometry.polygon import Polygon
+from shapely.geometry import Point
+import geopy
+from geopy.distance import geodesic
+import pyproj
+import getopt
+import sys
+
+polygon_width_meters = 100
+
+
+def main():
+
+    icone_file, cotrip_file, outputfile = parse_arguments(
+        sys.argv[1:], default_output_file_name='combined_wzdx_message.geojson')
+
+    if icone_file and cotrip_file:
+        try:
+            with open(outputfile, 'w+') as f:
+                wzdx_icone = json.loads(open(
+                    icone_file, 'r').read())
+                wzdx_cotrip = json.loads(open(
+                    cotrip_file, 'r').read())
+        except ValueError as e:
+            raise ValueError(
+                'One or more files specified are invalid. Please specify valid geojson files!') from None
+
+        polygon = generate_polygon(
+            wzdx_cotrip['features'][0]['geometry']['coordinates'], polygon_width_meters)
+        feature = iterate_feature(polygon, wzdx_icone)
+        if feature:
+
+            f.write(json.dumps(combine_wzdx(
+                wzdx_cotrip, wzdx_icone, feature), indent=2))
+            print(
+                'Combined WZDx message was written to combined_wzdx_message.geojson file.')
+
+        else:
+            print('no duplicate messages were found')
+    else:
+        print('please specify an input json file with -i and -c')
+        print(help_string)
+
+
+def combine_wzdx(wzdx_cotrip, wzdx_icone, icone_feature):
+    combined_wzdx = copy.deepcopy(wzdx_cotrip)
+    combined_wzdx['features'][0]['properties']['vehicle_impact'] = icone_feature['properties']['vehicle_impact']
+    combined_wzdx['road_event_feed_info']['data_sources'].extend(
+        wzdx_icone['road_event_feed_info']['data_sources'][0])
+
+    return combined_wzdx
+
+
+def iterate_feature(polygon, wzdx_message):
+    for feature in wzdx_message['features']:
+        for coord in feature['geometry']['coordinates']:
+            if isPointInPolygon(Point(coord[1], coord[0]), polygon):
+                return feature
+
+
+# generate polygon from list of geometry ([[long, lat], ...]) and width in meters
+def generate_polygon(geometry, polygon_width_in_meters):
+    """generate polygon from list of geometry ([[long, lat], ...]) as linestring and width in meters
+
+    Args: 
+        geometry: Linestring
+        polygon_width: width in meters
```

Comment:
```suggestion
        polygon_width_in_meters: width in meters
```
Update to match actual argument

Reply:
chaned it to polygon_width_in_meters